### ISSUE TITLE
Error while analysing 'lsl.w' instruction with relse engine

### ISSUE THREAD

**Issue by cchr-ledger** [binsec/binsec#30]
Hello, I am hitting an error while trying to analyse an arm32 function with the checkct plugin.

Here's a reduced test case binary
```console
Disassembly of section .text:

000200e4 <_start>:
   200e4: 80 ea 00 00   eor.w   r0, r0, r0
   200e8: 03 fa 0c fc   lsl.w   r12, r3, r12
   200ec: 83 ea 03 03   eor.w   r3, r3, r3
   200f0: fe de         trap
```
which, when analysed with the following script
```console
starting from <_start>
with concrete stack pointer
lr := 0x8badf00d ^ 1
halt at 0x8badf00d ^ 1
explore all
```
yields
```console
binsec -sse -checkct -sse-script checkct.cfg -arm-supported-modes thumb target
[sse:info] TTY: press [space] to switch between log and monitor modes.
[sse:error] Cut path 1 (uninterpreted "lsl.w\tip, r3, ip") @ 0x000200e8
[sse:info] Empty path worklist: halting ...
[sse:info] SMT queries
[...]
[checkct:result] Program status is : unknown (0.023)
[checkct:info] 0 visited path covering 2 instructions
[checkct:info] 0 / 0 control flow checks pass
[checkct:info] 0 / 0 memory access checks pass
[checkct:warning] Exploration is incomplete:
```
I am guessing this is an issue in unisim_archisec?

**Comment by recoules** [comment]
Hi @cchr-ledger,

Yes, this is also an issue related to the decoder.
It should be fixed (among other things) in the just released version [`0.0.7`](https://github.com/binsec/unisim_archisec/releases/tag/0.0.7).

Regard,
Frédéric

**Comment by cchr-ledger** [comment]
Perfect, thanks a lot!
